server/athenian/api/align/goals/dbaccess.py

```diff
@@ -54,6 +64,49 @@ async def delete_goal(account_id: int, goal_id: int, sdb_conn: Connection) -> No
     await sdb_conn.execute(sa.delete(Goal).where(where_clause))
 
 
+async def delete_team_goals(
+    account_id: int, goal_id: int, team_ids: Sequence[int], sdb_conn: Connection,
+) -> None:
+    """Delete a set of TeamGoal-s from DB."""
+    assert team_ids
+    assert conn_in_transaction(sdb_conn)
+    await _validate_team_goal_deletions(account_id, goal_id, team_ids, sdb_conn)
+
+    where_clause = sa.and_(TeamGoal.goal_id == goal_id, TeamGoal.team_id.in_(team_ids))
+    delete_stmt = sa.delete(TeamGoal).where(where_clause)
+    await sdb_conn.execute(delete_stmt)
+
+
+async def assign_team_goals(
+    account_id: int,
+    goal_id: int,
+    assignments: Sequence[TeamGoalTargetAssignment],
+    sdb_conn: Connection,
+) -> None:
+    """Assign new TeamGoal-s targets for an existing Goal."""
+    await _validate_team_goal_assignments(account_id, goal_id, assignments, sdb_conn)
+
+    now = datetime.now(timezone.utc)
+    values = [
+        {
+            "goal_id": goal_id,
+            "team_id": assign.team_id,
+            "target": assign.target,
+            "created_at": now,
+            "updated_at": now,
+        }
+        for assign in assignments
+    ]
+
+    insert = dialect_specific_insert(sdb_conn)
+    stmt = insert(TeamGoal)
+    upsert_stmt = stmt.on_conflict_do_update(
+        index_elements=["goal_id", "team_id"],
```

Comment:
Regarding the indexed elements: do we have an index for those two fields?

Reply:
yes this is the primary key of the table